TowersOfHanoi/TowersOfHanoi.cs

```diff
@@ -1,12 +1,270 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Threading;
 
 namespace TowersOfHanoi
 {
     class Program
     {
         static void Main(string[] args)
         {
-            Console.WriteLine("Hello World!");
+            // Instantiates a new game
+            Game game = new Game();
+            game.StartGame();
         }
     }
-}
+
+    class Game
+    {
+        // Declares a new dictionary that contains a String name for a tower in the key
+        // and a Tower object that contains a Stack of Block objects
+        Dictionary<String, Tower> gameboard;
+
+        // A string that changes from "from" to "to" as the player moves blocks from one tower
+        // to another tower.
+        String towerMove = "from";
+
+        int numberOfBlocks = 0;
+
+        public Game()
+        {
+            // Instantiates gameboard
+            gameboard = new Dictionary<String, Tower>();
+
+            Console.Clear();
+            numberOfBlocks = CreateGameBoard();
+
+            // Commented out 
+            // Populates the gameboard dictionary. Tower A having a Tower object with 4 Block objects
+            // and tower B and C being populated with empty Tower objects
+            // gameboard.Add("A", new Tower(4));
+            // gameboard.Add("B", new Tower());
+            // gameboard.Add("C", new Tower());
+        }
+
+        public void StartGame()
+        {
+            bool won = false;
+            // int block = 0;  Was passed an int from PickBlock method (legacy) no longer used to show thought process
+
+            Console.Clear();
+            DrawGameBoard();
+
+            while(!won)
+            {
+                String towerOrigin = "";
+                String towerDestination = "";
+
+                try
+                {
+                    towerOrigin = GetUserInput();
+                    // block = PickBlock(towerOrigin);  Feel this is less effective and creates more code to manage
+                    towerDestination = GetUserInput();
+                    MoveBlock(/* block,  */towerOrigin, towerDestination);
+                    Console.Clear();
+                    DrawGameBoard();
+                    won = CheckForWin();
+                }
+                catch
+                {
+                    Console.WriteLine("Invalid move.");
+                }
+            }
+
+            Console.WriteLine("You win!");
+        }
+
+        // Builds the gameboard. Asks the user how may blocks they would like to play with and builds the block
+        // objects in the first tower based on user input. There are a minimum of 4 blocks created.
+        public int CreateGameBoard()
+        {
+            Console.WriteLine("Please enter the number of blocks you want to play with if more than 4: ");
```

Comment:
I would have expected this question to be asked before the game is instantiated, and the value passed in to the game constructor